discovery_engine_core/web-api2/src/elastic.rs

```diff
@@ -123,3 +213,15 @@ impl ElasticSearchClient {
         }
     }
 }
+
+#[derive(Deserialize)]
+struct BulkResponse {
+    errors: bool,
+    items: Vec<HashMap<String, BulkItemResponse>>,
+}
+
+#[derive(Debug, Deserialize)]
+struct BulkItemResponse {
+    status: u16,
+    error: Value,
```

Comment:
is error always in the response? I know the ingestion request we used an Option there
```suggestion
    error: Option<Value>,
```

Reply:
yes, through it's only missing a `#[serde(default)]` but we don't need to wrap it into an option as there is no semantic difference between not given and set to `null`.